demo/src/main/java/org/ode4j/demo/DemoMotor.java

```diff
@@ -45,76 +45,67 @@
-	private static DLMotorJoint[] lmotor = new DLMotorJoint[2];
-	private static DAMotorJoint[] amotor = new DAMotorJoint[2];
+	private static final DBody[] body = new DBody[2];
+	private static final DBox[] geom = new DBox[2];
+	private static final DLMotorJoint[] lmotor = new DLMotorJoint[2];
+	private static final DAMotorJoint[] amotor = new DAMotorJoint[2];
 	private static DSpace space;
 	private static DJointGroup contactgroup;
 
-
-	// start simulation - set viewpoint
-
+	// viewpoint at start
 	private static float[] xyz = {1.0382f,-1.0811f,1.4700f};
 	private static float[] hpr = {135.0000f,-19.5000f,0.0000f};
+
 	@Override
-	public void start()
-	{
+	public void start() {
 		dsSetViewpoint (xyz,hpr);
 		System.out.println ("Press 'q,a,z' to control one axis of lmotor connectiong two bodies. (q is +,a is 0, z is -)");
-		System.out.println ("Press 'w,e,r' to control one axis of lmotor connectiong first body with world. (w is +,e is 0, r is -)");
+		System.out.println ("Press 'w,e,r' to control one axis of amotor connectiong first body with world. (w is +,e is 0, r is -)");
 	}
 
-
 	// called when a key pressed
 	@Override
-	public void command (char cmd)
-	{
-		if (cmd == 'q' || cmd == 'Q') {
-			lmotor[0].setParamVel(0);
-			lmotor[0].setParamVel2(0);
-			lmotor[0].setParamVel3(0.1);
-		} else if (cmd == 'a' || cmd == 'A') {
-			lmotor[0].setParamVel(0);
-			lmotor[0].setParamVel2(0);
-			lmotor[0].setParamVel3(0);
-		} else if (cmd == 'z' || cmd == 'Z') {
-			lmotor[0].setParamVel(0);
-			lmotor[0].setParamVel2(0);
-			lmotor[0].setParamVel3(-10.1);
-		} else if (cmd == 'w' || cmd == 'W') {
-			lmotor[1].setParamVel(10.1);
-			lmotor[1].setParamVel2(0);
-			lmotor[1].setParamVel3(0);
-		} else if (cmd == 'e' || cmd == 'E') {
-			lmotor[1].setParamVel(0);
-			lmotor[1].setParamVel2(0);
-			lmotor[1].setParamVel3(0);
-		} else if (cmd == 'r' || cmd == 'R') {
-			lmotor[1].setParamVel(-10.1);
-			lmotor[1].setParamVel2(0);
-			lmotor[1].setParamVel3(0);
+	public void command (char cmd) {
+		// get the forces
+		double linear = lmotor[0].getParam(DJoint.PARAM_N.dParamVel3);
+		double angular = amotor[1].getParam(DJoint.PARAM_N.dParamVel1);
+		// change the forces based on the key
+		switch(cmd) {
+		case 'q': case 'Q':  linear += ACCELERATION;  break;
+		case 'a': case 'A':  linear =0;  break;
+		case 'z': case 'Z':  linear -= ACCELERATION;  break;
+		case 'w': case 'W':  angular += ACCELERATION;  break;
+		case 'e': case 'E':  angular =0;  break;
+		case 'r': case 'R':  angular -= ACCELERATION;  break;
+		default:  break;
 		}
+		// report the new forces
+		System.out.println("linear="+linear+" angular="+angular);
+		// apply the new forces
+		lmotor[0].setParamVel3(linear);
+		amotor[1].setParamVel(angular);
 	}
 
-
-
-	private static void nearCallback (Object data, DGeom o1, DGeom o2)
-	{
+	private void nearCallback (Object data, DGeom o1, DGeom o2) {
 		// exit without doing anything if the two bodies are connected by a joint

```

Comment:
This comment is misleading.  Was it supposed to be a TODO?

Reply:
The comment is not from me. I will have a look.